overlord/restart/restart.go

```diff
@@ -99,6 +100,7 @@ func Manager(st *state.State, curBootID string, h Handler) (*RestartManager, err
 	}
 
 	st.RegisterPendingChangeByAttr("wait-for-system-restart", rm.pendingForSystemRestart)
+	rm.changeCallbackID = st.AddChangeStatusChangedHandler(processRestartForChange)
```

Comment:
the callback is set but never used

Reply:
True, what's your thoughts here? I'm thinking about adding the `Stop()` call and making sure we remove the handler again. Would that be fine?

Reply:
that's not unreasonable